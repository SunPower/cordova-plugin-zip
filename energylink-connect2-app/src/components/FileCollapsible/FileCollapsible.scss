@import '~@sunpower/theme-dark/styles/variables';

.sp-color-sun {
  color: $sun;
}
.firmware-screen {
  .button {
    letter-spacing: initial;
  }
}
