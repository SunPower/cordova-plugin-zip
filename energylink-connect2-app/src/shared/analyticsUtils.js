import {
  __,
  compose,
  curry,
  gt,
  isEmpty,
  join,
  lensIndex,
  map,
  not,
  over,
  props,
  propSatisfies,
  toLower,
  toUpper,
  when
} from 'ramda'

export const isNotEmpty = compose(not, isEmpty)

export const capitalizeWord = when(
  propSatisfies(gt(__, 0), 'length'),
  compose(join(''), over(lensIndex(0), toUpper), toLower)
)

const stringValueOrEmptyString = stringValue => stringValue || ''

const getStringProps = curry(compose(map(stringValueOrEmptyString), props))

export const getUserProfile = getStringProps([
  'uniqueId',
  'firstName',
  'lastName',
  'email',
  'dealerName',
  'recordType'
])
