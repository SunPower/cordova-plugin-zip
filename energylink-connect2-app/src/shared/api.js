import Swagger from 'swagger-client'
let apiParty, apiSite, apiFirmware, apiScanSN

const requestOptions = access_token => ({
  requestInterceptor: req => {
    if (access_token) req.headers['Authorization'] = `Bearer ${access_token}`
    return req
  }
})

export async function getApiFirmware() {
  if (!apiFirmware)
    apiFirmware = await Swagger(process.env.REACT_APP_SWAGGER_FIRMWARE)
  return apiFirmware
}
export async function getApiPVS() {
  return await Swagger(process.env.REACT_APP_PVS_ADDRESS)
}
export async function getApiDevice(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_DEVICE,
    requestOptions(access_token)
  )
}
export async function getApiParty(access_token) {
  apiParty = await Swagger(
    process.env.REACT_APP_SWAGGER_PARTY,
    requestOptions(access_token)
  )
  return apiParty
}
export async function getApiSearch(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_SEARCH,
    requestOptions(access_token)
  )
}
export async function getApiSite(access_token) {
  if (!apiSite)
    apiSite = await Swagger(
      process.env.REACT_APP_SWAGGER_SITE,
      requestOptions(access_token)
    )
  return apiSite
}
export async function getApiScanSN() {
  if (!apiScanSN)
    apiScanSN = await Swagger(process.env.REACT_APP_SCAN_SERIAL_NUMBERS)
  return apiScanSN
}
export async function getApiAuth(access_token) {
  return await Swagger(
    process.env.REACT_APP_SWAGGER_AUTH,
    requestOptions(access_token)
  )
}
export const storageSwaggerTag = 'Commissioning'
