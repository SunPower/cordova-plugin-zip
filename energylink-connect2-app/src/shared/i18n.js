import { useMemo } from 'react'
import { useSelector } from 'react-redux'

import * as languages from './locales'
import * as localizedComponents from './localizedComponents'

const getTranslate = translations => (key, ...params) => {
  let value = translations[key] || key
  if (params) {
    params.forEach((p, idx) => {
      value = value.replace(`{${idx}}`, p)
    })
  }
  return value
}

export const useI18n = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language
  return useMemo(() => getTranslate(languages[locale]), [locale])
}

export const useI18nComponent = () => {
  const language = useSelector(state => state.language)
  const { locale = 'en' } = language || {}

  return useMemo(
    () => componentName => {
      const path = `${locale}_${componentName}`
      const Component = localizedComponents[path] || {}
      if (typeof Component !== 'function') {
        throw new Error(`Component not found: ${path}`)
      }
      return Component()
    },
    [locale]
  )
}
