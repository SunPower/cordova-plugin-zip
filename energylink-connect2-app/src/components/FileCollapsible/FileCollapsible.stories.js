import { storiesOf } from '@storybook/react'
import React from 'react'

import FileCollapsible from './index'

storiesOf('File collapsible', module).add('Downloading', () => (
  <div className="full-min-height pl-10 pr-10">
    <FileCollapsible
      fileName="File downloading"
      progress={20}
      step="DOWNLOADING"
      isDownloading
    />
  </div>
))
storiesOf('File collapsible', module).add('Saving file', () => (
  <div className="full-min-height pl-10 pr-10">
    <FileCollapsible
      fileName="Writing file"
      step="WRITING_FILE"
      isDownloading
      progress={40}
    />
  </div>
))

storiesOf('File collapsible', module).add('Downloaded', () => (
  <div className="full-min-height pl-10 pr-10">
    <FileCollapsible
      size={40.3}
      fileName="File completely downloaded"
      isDownloaded={true}
    />
  </div>
))

storiesOf('File collapsible', module).add('Error', () => (
  <div className="full-min-height pl-10 pr-10">
    <FileCollapsible fileName="With Error" error="something happened" />
  </div>
))
