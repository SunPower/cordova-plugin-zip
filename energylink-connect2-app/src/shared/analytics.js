import appVersion from '../macros/appVersion.macro'
import { capitalizeWord, getUserProfile } from 'shared/analyticsUtils'
import { MIXPANEL_EVENT_QUEUED } from 'state/actions/analytics'
import { getAppleDeviceFamily } from 'shared/appleDevicesTable'
import { prop } from 'ramda'

const getAppAndDeviceProperties = () => {
  const { device } = window
  const { model, version, platform, manufacturer } = device

  let deviceModel
  let deviceBrowserVersion

  if (platform === 'iOS') {
    deviceModel = getAppleDeviceFamily(model)
    deviceBrowserVersion = version
  } else {
    deviceModel = `${manufacturer}, ${model}`
  }

  return {
    'App Build': appVersion(),
    'App Flavor': process.env.REACT_APP_FLAVOR,
    'OS Version': version,
    ...(deviceBrowserVersion ? { $browser_version: deviceBrowserVersion } : {}),
    $device: deviceModel
  }
}

export const loggedIn = user => {
  const { mixpanel } = window
  const {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  } = getUserProfile(user)
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Login', { Success: true })

  return MIXPANEL_EVENT_QUEUED('Login - Success')
}

export const loginFailed = () => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.track('Login', { Success: false })

  return MIXPANEL_EVENT_QUEUED('Login - Failed')
}

export const scanPVS = ({ pvsSN, entryMethod }, event) => {
  const { mixpanel } = window

  mixpanel.register({ 'PVS SN': pvsSN })
  mixpanel.track('Scan PVS Tag', {
    'Entry Method': entryMethod
  })

  return MIXPANEL_EVENT_QUEUED(event)
}

export const siteNotFound = ({
  uniqueId: userId,
  firstName,
  lastName,
  email,
  dealerName,
  recordType: dealerType
}) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', { Found: false })

  return MIXPANEL_EVENT_QUEUED('Find Site - site not found')
}

export const siteFound = (
  {
    uniqueId: userId,
    firstName,
    lastName,
    email,
    dealerName,
    recordType: dealerType
  },
  { city, st_id, postalCode, siteKey, commissioned }
) => {
  const { mixpanel } = window
  const appAndDeviceProperties = getAppAndDeviceProperties()

  mixpanel.register(appAndDeviceProperties)
  mixpanel.identify(userId)
  mixpanel.people.set(appAndDeviceProperties)
  mixpanel.people.set({
    $first_name: firstName,
    $last_name: lastName,
    $email: email,
    'Dealer Name': dealerName,
    'Dealer Type': capitalizeWord(dealerType)
  })
  mixpanel.track('Find Site', {
    Found: true,
    ...{
      $city: city,
      State: st_id,
      'Zip Code': postalCode,
      'Site ID': siteKey,
      Commissioned: commissioned
    }
  })

  return MIXPANEL_EVENT_QUEUED('Find Site - site found')
}

export const saveConfiguration = config => {
  const { mixpanel } = window
  mixpanel.track('Configure', config)
  return MIXPANEL_EVENT_QUEUED('Configure - submit site config')
}

export const saveInventory = (inventory, hasEss) => {
  const { mixpanel } = window

  mixpanel.register({
    'Equinox Storage Site': hasEss
  })

  mixpanel.track('Inventory', {
    'AC Module Count': prop('AC_MODULES', inventory),
    'ESS System Size': hasEss ? prop('ESS', inventory) : 'None',
    'Equinox Storage Site': hasEss
  })
  return MIXPANEL_EVENT_QUEUED('Inventory')
}

export const deviceResume = pageResume => {
  const { mixpanel } = window
  mixpanel.register({
    Page: pageResume
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

/**
 *
 * @param {string}[interfaceType] - The type of connection the user has. either can be  PVS, Cell, or Customer WiFi
 * @return {Action<null, null>}
 */
export const internetConnection = (interfaceType = '') => {
  const { mixpanel } = window
  mixpanel.register({
    'Internet Connection': interfaceType
  })
  return MIXPANEL_EVENT_QUEUED('User opens app - Register page resume')
}

export const firmwareUpdateEvent = data => {
  const { mixpanel } = window
  mixpanel.track('Firmware Update', {
    'Current FW Version': prop('fromFWVersion', data),
    'Current BN Version': prop('fromBuildNumber', data),
    'New FW Version': prop('toFWVersion', data),
    'New BN Version': prop('toBuildNumber', data),
    Success: prop('success', data)
  })
  return MIXPANEL_EVENT_QUEUED('Firmware update - Update successful')
}

export const timeMixPanelEvent = name => {
  if (!name) throw new Error('The name parameter is required')
  const { mixpanel } = window
  mixpanel.time_event(name)
}

export const commissionSite = ({ duration, timeConfiguring }) => {
  const { mixpanel } = window
  mixpanel.track('Commission Site', {
    $duration: duration,
    'Time Configuring': timeConfiguring
  })
  return MIXPANEL_EVENT_QUEUED(
    'Commission Site - first successful site configuration'
  )
}
