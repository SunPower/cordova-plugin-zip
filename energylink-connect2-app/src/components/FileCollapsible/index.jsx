import clsx from 'clsx'
import React from 'react'
import { useI18n } from 'shared/i18n'
import Collapsible from 'components/Collapsible'
import { either } from 'shared/utils'
import { Link } from 'react-router-dom'

const FileCollapsible = ({
  progress = 0,
  error = null,
  fileName = '',
  isDownloading = false,
  isDownloaded = false,
  version = '',
  downloadFile,
  step = '',
  size = 10,
  goToReleaseNotes = null
}) => {
  const t = useI18n()
  const showReleaseNotes = isDownloaded && goToReleaseNotes

  return (
    <Collapsible
      title={fileName}
      actions={either(
        !isDownloading && (error || isDownloaded),
        <span
          className={clsx('is-size-4', {
            'sp-download': !isDownloaded,
            'sp-update': isDownloaded
          })}
          onClick={downloadFile}
        />
      )}
      expanded
    >
      {either(
        error,
        <span>{t('FIRMWARE_ERROR_FOUND')}</span>,
        <>
          {either(
            version,
            <section className="mb-5">
              <span className="has-text-white">
                {t('VERSION')}: {version}
              </span>
            </section>
          )}
          <section className="mt-20 mb-10">
            <p className="mb-5">
              <span className="mr-10 has-text-white has-text-weight-bold">
                {either(isDownloading, progress, 100)}%
              </span>
              {either(
                isDownloading,
                t(step),
                !isDownloaded ? t('NOT_DOWNLOADED') : t('DOWNLOADED')
              )}
              {either(
                size,
                <span className="is-pulled-right has-text-white has-text-weight-bold">
                  {size} MB
                </span>
              )}
            </p>
            {either(
              isDownloading,
              <progress
                className="progress is-tiny is-white"
                value={progress}
                max="100"
              />
            )}
          </section>
          {either(
            showReleaseNotes,
            <section className="mb-5">
              <Link className="has-text-weight-bold" onClick={goToReleaseNotes}>
                {t('RELEASE_NOTES')}
              </Link>
            </section>
          )}
        </>
      )}
    </Collapsible>
  )
}
export default FileCollapsible
