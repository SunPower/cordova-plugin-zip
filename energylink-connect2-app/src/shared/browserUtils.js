import { createExternalLinkHandler } from 'shared/routing'

export function openUrl(url) {
  window.SafariViewController.isAvailable(function(available) {
    if (available) {
      window.SafariViewController.show({
        url: url,
        hidden: false, // default false. You can use this to load cookies etc in the background (see issue #1 for details).
        animated: false, // default true, note that 'hide' will reuse this preference (the 'Done' button will always animate though)
        transition: 'curl', // (this only works in iOS 9.1/9.2 and lower) unless animated is false you can choose from: curl, flip, fade, slide (default)
        enterReaderModeIfAvailable: false, // default false
        tintColor: '#ffffff', // default is ios blue
        barColor: '#15202e', // on iOS 10+ you can change the background color as well
        controlTintColor: '#15202e' // on iOS 10+ you can override the default tintColor
      })
    } else {
      // potentially powered by InAppBrowser because that (currently) clobbers window.open
      createExternalLinkHandler(url)()
    }
  })
}

export function dismissBrowser() {
  window.SafariViewController.hide()
}
