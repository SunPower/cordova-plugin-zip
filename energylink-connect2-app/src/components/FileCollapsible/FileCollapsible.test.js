import { shallow } from 'enzyme'
import React from 'react'

import FileCollapsible from './index'

import * as i18n from 'shared/i18n'

describe('File Collapsible component', () => {
  beforeEach(function() {
    jest.spyOn(i18n, 'useI18n').mockImplementation(() => (key, ...params) => {
      return `${key.toUpperCase()} ${params.join('_')}`.trim()
    })
  })
  test('renders correctly with no props', () => {
    const component = shallow(<FileCollapsible />)
    expect(component).toMatchSnapshot()
  })
  test('renders correctly while downloading', () => {
    const component = shallow(
      <FileCollapsible
        progress={80}
        fileName="test fileName"
        size={40.0}
        isDownloading
        step="DOWNLOADING"
      />
    )
    expect(component).toMatchSnapshot()
  })
  test('renders correctly complete', () => {
    const component = shallow(
      <FileCollapsible
        progress={100}
        fileName="test fileName"
        size={40.0}
        isDownloaded
        step="COMPLETE"
      />
    )
    expect(component).toMatchSnapshot()
  })
})
