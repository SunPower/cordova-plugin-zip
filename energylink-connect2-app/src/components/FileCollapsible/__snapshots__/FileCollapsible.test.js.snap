// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File Collapsible component renders correctly complete 1`] = `
<Collapsible
  actions={
    <span
      className="is-size-4 sp-color-sun sp-update"
      onClick={[Function]}
    />
  }
  expanded={true}
  title="test fileName"
>
  <section
    className="mb-5"
  >
    <span
      className="has-text-white"
    >
      YOU_HAVE_THE_LATEST_FIRMWARE test fileName
    </span>
  </section>
</Collapsible>
`;

exports[`File Collapsible component renders correctly while downloading 1`] = `
<Collapsible
  actions={null}
  expanded={true}
  title="test fileName"
>
  <section
    className="mb-5"
  >
    <span
      className="has-text-white"
    />
  </section>
  <section
    className="mt-20 mb-10"
  >
    <p
      className="mb-5"
    >
      <span
        className="mr-10 has-text-white has-text-weight-bold"
      >
        80
        %
      </span>
      DOWNLOADING
      <span
        className="is-pulled-right has-text-white has-text-weight-bold"
      >
        40
         MB
      </span>
    </p>
    <progress
      className="progress is-tiny is-white"
      max="100"
      value={80}
    />
  </section>
</Collapsible>
`;

exports[`File Collapsible component renders correctly with no props 1`] = `
<Collapsible
  actions={
    <span
      className="is-size-4 sp-color-sun sp-download"
      onClick={[Function]}
    />
  }
  expanded={true}
  title=""
>
  <section
    className="mb-5"
  >
    <span
      className="has-text-white"
    >
      <span
        className="has-text-white has-text-weight-bold"
      >
        CHECKING_LATEST_FIRMWARE_VERSION
      </span>
    </span>
  </section>
</Collapsible>
`;
